ddprof-ffi/src/exporter.rs

```diff
@@ -277,6 +285,33 @@ pub unsafe extern "C" fn profile_exporter_send(
     }
 }
 
+/// Can be passed as an argument to send and then be used to asynchronously cancel it from a different thread.
+#[no_mangle]
+#[must_use]
+pub extern "C" fn ddprof_ffi_CancellationToken_new() -> Box<tokio_util::sync::CancellationToken> {
+    Box::new(tokio_util::sync::CancellationToken::new())
+}
+
+/// Cancel request with the given token. Can only be called once for a given token.
+/// Returns `true` if token was successfully cancelled.
+#[no_mangle]
+pub extern "C" fn ddprof_ffi_CancellationToken_cancel(cancel: Option<Box<tokio_util::sync::CancellationToken>>) -> bool {
+    let cancel_ptr = match cancel {
+        Some(cancel) => cancel,
+        None => return false,
+    };
+
+    if cancel_ptr.is_cancelled() { return false }
+    cancel_ptr.cancel();
+
+    return true
+}
+
+#[no_mangle]
+pub extern "C" fn ddprof_ffi_CancellationToken_drop(cancel: Option<Box<tokio_util::sync::CancellationToken>>) {
+    std::mem::drop(cancel)
+}
```

Comment:
Thanks for the suggestion! I saw you use that pattern and half-suspected it may be doing what you claimed, but I wasn't entirely sure until now.

Fixed :)